### PR TITLE
Visitor can filter articles based on CATEGORIES

### PR DESCRIPTION
**Pull request by leiter007**
# [PT Story](https://www.pivotaltracker.com/story/show/166152055)

## Changes proposed in this pull request:
* Integration test for filtering articles by category via header category tabs
* Added component for ArticlesByCategory which filters articles on category - either by clicking in the tabs or directly from the URL path
* Refactors paths in App.js to redirect to ArticlesByCategory component for all articles, including landing page   
* Removes ListOfArticles component since not needed anymore

## What I have learned working on this feature:
* Changing states based on ComponentDidUpdate
* Props, states and inheritance...

## Screenshot:

Screenshot 2019-06-04 at 14 41 57

Screenshot 2019-06-04 at 14 41 10

Screenshot 2019-06-04 at 14 42 07

Screenshot 2019-06-05 at 11 56 39

### CODE REVIEW

**Review by Carrosen — COMMENTED** [review]
Awesome guys

File: src/index.js [hunk 1]
```diff
@@ -6,11 +6,13 @@ import * as serviceWorker from './serviceWorker';
 import { BrowserRouter } from 'react-router-dom'
 import axios from 'axios'
 
-axios.defaults.baseURL = 'https://glocal-news.herokuapp.com'
+axios.defaults.baseURL = 'http://localhost:3002'
```
Comment (raptorf1) [inline src/index.js:13]: back to heroku please :P

**Review by raptorf1 — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 1 above)
Comment (leiter007) [reply src/index.js:13]: Ah yes, done! :) Thank you

File: src/Components/HeaderMain.jsx [hunk 2]
```diff
@@ -34,76 +34,92 @@ const loggedOutLabels = ['Sign Up', 'Log In']
...
-            as={Link} 
-            to='/'
-            style={{ fontSize: "2em" }}
-          >
-            GLOCAL NEWS
-          </Header>
-        <Divider hidden />
-      </Container>
+class HeaderMain extends Component {
+state = {
```
Comment (oliverochman) [inline src/Components/HeaderMain.jsx]: indentation

Comment (leiter007) [reply src/Components/HeaderMain.jsx]: Done

File: src/Components/HeaderMain.jsx [hunk 3]
```diff
@@ -1,13 +1,78 @@
...
-import { Menu, Header, Select, Container, Divider, Segment } from 'semantic-ui-react';
-import { Link } from 'react-router-dom';
+import HeaderCategory from './HeaderCategory'
+import { Menu, Header, Select, Container, Divider, Segment } from 'semantic-ui-react'
+import { Link } from 'react-router-dom'
 import { connect } from 'react-redux';
 import { signOutUser } from '../reduxTokenAuthConfig';
 import { withRouter } from 'react-router-dom';
 
+// const countryOptions = [
```
Comment (oliverochman) [inline src/Components/HeaderMain.jsx]: why is this commented out?

**Review by oliverochman — CHANGES_REQUESTED** [review]

File: src/Components/HeaderMain.jsx (hunk 3 above)
Comment (raptorf1) [reply src/Components/HeaderMain.jsx]: done

**Review by oliverochman — APPROVED** [review]

**Pull request merged**